cmd/oras/repository/tags.go

```diff
@@ -78,14 +85,37 @@ func showTags(opts showTagsOptions) error {
 	if err != nil {
 		return err
 	}
+	filter := ""
 	if opts.Reference != "" {
-		return fmt.Errorf("unexpected tag or digest %q found in repository reference %q", opts.Reference, opts.RawReference)
+		_, err := digest.Parse(opts.Reference)
+		if err == nil {
+			filter = opts.Reference
+		} else {
+			desc, err := finder.Resolve(ctx, opts.Reference)
+			if err != nil {
+				return err
+			}
+			filter = desc.Digest.String()
+		}
+		fmt.Fprintln(os.Stderr, "Tag query by reference may take a while")
```

Comment:
You may get the `logger` from `opts.SetLoggerLevel()` 
```suggestion
		logger.Infof("[Preview] Querying tags associated to %s, it may take a while.", filter)
```

Reply:
Does that log to stderr?

Reply:
Yeh, that works

Reply:
Yes, it logs to stderr.

Reply:
Done